server/src/main/java/org/opensearch/index/compositeindex/datacube/startree/builder/StarTreeDocValuesIteratorAdapter.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.index.compositeindex.datacube.startree.builder;
+
+import org.apache.lucene.codecs.DocValuesProducer;
+import org.apache.lucene.index.DocValuesType;
+import org.apache.lucene.index.FieldInfo;
+import org.apache.lucene.index.SortedNumericDocValues;
+import org.apache.lucene.search.DocIdSetIterator;
+import org.opensearch.common.annotation.ExperimentalApi;
+import org.opensearch.index.compositeindex.datacube.startree.utils.SequentialDocValuesIterator;
+
+import java.io.IOException;
+
+/**
+ * A factory class to return respective doc values iterator based on the doc volues type.
+ *
+ * @opensearch.experimental
+ */
+@ExperimentalApi
+public class StarTreeDocValuesIteratorAdapter {
+
+    /**
+     * Creates an iterator for the given doc values type and field using the doc values producer
+     */
+    public SequentialDocValuesIterator getDocValuesIterator(DocValuesType type, FieldInfo field, DocValuesProducer producer)
+        throws IOException {
+        switch (type) {
+            case SORTED_NUMERIC:
+                return new SequentialDocValuesIterator(producer.getSortedNumeric(field));
+            default:
+                throw new IllegalArgumentException("Unsupported DocValuesType: " + type);
+        }
+    }
+
+    /**
+     * Returns the next value for the given iterator
+     */
+    public Long getNextValue(SequentialDocValuesIterator sequentialDocValuesIterator, int currentDocId) throws IOException {
+        if (sequentialDocValuesIterator.getDocIdSetIterator() instanceof SortedNumericDocValues) {
```

Comment:
Can we move this to iterator ?

Reply:
yeah, this will be taken in the subsequent PR